jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/NBuildStepsRunner.java

```diff
@@ -0,0 +1,270 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.cloud.tools.jib.api.Containerizer;
+import com.google.cloud.tools.jib.api.JibContainer;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.builder.BuildSteps;
+import com.google.cloud.tools.jib.configuration.CacheDirectoryCreationException;
+import com.google.cloud.tools.jib.configuration.LayerConfiguration;
+import com.google.cloud.tools.jib.event.EventDispatcher;
+import com.google.cloud.tools.jib.event.events.LogEvent;
+import com.google.cloud.tools.jib.image.ImageReference;
+import com.google.cloud.tools.jib.image.LayerEntry;
+import com.google.cloud.tools.jib.registry.InsecureRegistryException;
+import com.google.cloud.tools.jib.registry.RegistryAuthenticationFailedException;
+import com.google.cloud.tools.jib.registry.RegistryCredentialsNotSentException;
+import com.google.cloud.tools.jib.registry.RegistryErrorException;
+import com.google.cloud.tools.jib.registry.RegistryUnauthorizedException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Verify;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.concurrent.ExecutionException;
+import org.apache.http.conn.HttpHostConnectException;
+
+// TODO: This should replace BuildStepsRunner once jib-maven-plugin is changed to use this.
+/** Runs a {@link BuildSteps} and builds helpful error messages. */
+public class NBuildStepsRunner {
+
+  private static final String STARTUP_MESSAGE_PREFIX_FOR_DOCKER_REGISTRY =
+      "Containerizing application to ";
+  private static final String SUCCESS_MESSAGE_PREFIX_FOR_DOCKER_REGISTRY =
+      "Built and pushed image as ";
+
+  private static final String STARTUP_MESSAGE_PREFIX_FOR_DOCKER_DAEMON =
+      "Containerizing application to Docker daemon as ";
+  private static final String SUCCESS_MESSAGE_PREFIX_FOR_DOCKER_DAEMON =
+      "Built image to Docker daemon as ";
+
+  private static final String STARTUP_MESSAGE_FORMAT_FOR_TARBALL =
+      "Containerizing application to file at '%s'...";
+  private static final String SUCCESS_MESSAGE_FORMAT_FOR_TARBALL =
+      "Built image tarball at \u001B[36m%s\u001B[0m";
```

Comment:
Just asking: I feel like I've seen this class or most part of the code. Is this class just copied or a completely new one? If copied, is it possible to remove the original one (basically moving the class)?

Reply:
Yes it is copied. #1124 will remove the original.